### PR TITLE
Public API functions implementation

### PR DESCRIPTION
**Pull request by ZigzagAK**
Please review my patch. Thank you.

### CODE REVIEW

File: src/ngx_stream_lua_api.c [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+{
+    ngx_stream_session_t    *s;
+
+    lua_getglobal(L, ngx_stream_lua_session_key);
+    s = lua_touserdata(L, -1);
+    lua_pop(L, 1);
+
+    return s;
+}
+
```
Comment (agentzh) [inline src/ngx_stream_lua_api.c]: Style: two blank lines required here.

File: src/ngx_stream_lua_util.h [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
 
 #include "ngx_stream_lua_common.h"
-
+#include "api/ngx_stream_lua_api.h"
```
Comment (agentzh) [inline src/ngx_stream_lua_util.h]: Ditto.

File: src/ngx_stream_lua_api.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+lua_State *
+ngx_stream_lua_get_global_state(ngx_conf_t *cf)
+{
+    ngx_stream_lua_main_conf_t *lmcf;
+
+    lmcf = ngx_stream_conf_get_module_main_conf(cf, ngx_stream_lua_module);
+
+    return lmcf->lua;
+}
+
```
Comment (agentzh) [inline src/ngx_stream_lua_api.c]: Style: two blank lines required here.

Comment (ZigzagAK) [inline src/ngx_stream_lua_api.c]: completed

File: src/ngx_stream_lua_api.c (hunk 1 above)
Comment (ZigzagAK) [inline src/ngx_stream_lua_api.c]: completed

**Comment by ZigzagAK** [comment]
Ready.

**Comment by ZigzagAK** [comment]
Hi!
This pull request have a chance to be merged into mainstream?

**Comment by agentzh** [comment]
@ZigzagAK Yeah sure. It's just that we're planning a thorough refactoring in this module in the next couple of months and we'll take care of this part in the new rewrite. And we don't want to touch the current old code base any more. Thanks for your understanding!

**Comment by ZigzagAK** [comment]
@agentzh Thanks.

**Comment by dndx** [comment]
No longer needed.

**Pull request closed without merging**